sdk/eventgrid/Azure.Messaging.EventGrid/src/EventGridPublisherClient.cs

```diff
@@ -220,19 +220,50 @@ public virtual Response SendEvent(CloudEvent cloudEvent, CancellationToken cance
         /// <param name="cloudEvents"> The set of events to be published to Event Grid. </param>
         /// <param name="cancellationToken"> An optional cancellation token instance to signal the request to cancel the operation.</param>
         public virtual async Task<Response> SendEventsAsync(IEnumerable<CloudEvent> cloudEvents, CancellationToken cancellationToken = default)
-            => await SendCloudEventsInternal(cloudEvents, true /*async*/, cancellationToken).ConfigureAwait(false);
+            => await SendCloudEventsInternal(cloudEvents, null, true /*async*/, cancellationToken).ConfigureAwait(false);
 
         /// <summary> Publishes a set of CloudEvents to an Event Grid topic. </summary>
         /// <param name="cloudEvents"> The set of events to be published to Event Grid. </param>
         /// <param name="cancellationToken"> An optional cancellation token instance to signal the request to cancel the operation.</param>
         public virtual Response SendEvents(IEnumerable<CloudEvent> cloudEvents, CancellationToken cancellationToken = default)
-            => SendCloudEventsInternal(cloudEvents, false /*async*/, cancellationToken).EnsureCompleted();
+            => SendCloudEventsInternal(cloudEvents, null, false /*async*/, cancellationToken).EnsureCompleted();
+
+        /// <summary> Publishes a CloudEvent to an Event Grid topic. </summary>
+        /// <param name="cloudEvent"> The set of events to be published to Event Grid.</param>
+        /// <param name="options">The set of options to use when sending the CloudEvent.</param>
+        /// <param name="cancellationToken"> An optional cancellation token instance to signal the request to cancel the operation.</param>
+        [ForwardsClientCalls]
```

Comment:
That's new to me.   What does `ForwardsClientCalls` do?

Reply:
This is used to let the DiagnosticScope validation know that the scope will be created in a subsequent method that this calls into.